File: device-app/ios-object-c/helloUms/helloUms/bridgeUms/PushAppProto.m
#import "UmsProtocol.h"
#import "PushAppProto.h"

#pragma mark - PushAppAuthNumberReq

@implementation PushAppAuthNumberReq

- (instancetype)init {
    _cId = APP_AUTH_NUMBER_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    _cId = [dict[@"cId"] intValue];
    _aId = dict[@"aId"] ?: @"";
    _cc = dict[@"cc"] ?: @"";
    _pn = dict[@"pn"] ?: @"";
    return self;
}

- (NSDictionary *)toDictionary {
    return @{ @"cId": @(_cId), @"aId": _aId ?: @"", @"cc": _cc ?: @"", @"pn": _pn ?: @"" };
}

@end

#pragma mark - PushAppAuthNumberRes

@implementation PushAppAuthNumberRes

- (instancetype)init {
    _cId = APP_AUTH_NUMBER_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    _cId = [dict[@"cId"] intValue];
    _aId = dict[@"aId"] ?: @"";
    _rc = [dict[@"rc"] intValue];
    _cmt = dict[@"cmt"];
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [@{ @"cId": @(_cId), @"aId": _aId ?: @"", @"rc": @(_rc) } mutableCopy];
    if (_cmt != nil) dict[@"cmt"] = _cmt;
    return dict;
}

@end

#pragma mark - PushAppVerifyAuthNumberReq

@implementation PushAppVerifyAuthNumberReq

- (instancetype)init {
    _cId = APP_VERITY_AUTH_NUMBER_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    _cId = [dict[@"cId"] intValue];
    _aId = dict[@"aId"] ?: @"";
    _dt = [dict[@"dt"] intValue];
    _pn = dict[@"pn"] ?: @"";
    _an = dict[@"an"] ?: @"";
    return self;
}

- (NSDictionary *)toDictionary {
    return @{ @"cId": @(_cId), @"aId": _aId ?: @"", @"dt": @(_dt), @"pn": _pn ?: @"", @"an": _an ?: @"" };
}

@end

#pragma mark - PushAppVerifyAuthNumberRes

@implementation PushAppVerifyAuthNumberRes

- (instancetype)init {
    _cId = APP_VERITY_AUTH_NUMBER_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    _cId = [dict[@"cId"] intValue];
    _aId = dict[@"aId"] ?: @"";
    _rc = [dict[@"rc"] intValue];
    _cmt = dict[@"cmt"];
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [@{ @"cId": @(_cId), @"aId": _aId ?: @"", @"rc": @(_rc) } mutableCopy];
    if (_cmt != nil) dict[@"cmt"] = _cmt;
    return dict;
}

@end

#pragma mark - PushAppInstallReq

@implementation PushAppInstallReq

- (instancetype)init {
    _cId = APP_INSTALL_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    _cId = [dict[@"cId"] intValue];
    _aId = dict[@"aId"] ?: @"";
    _dt = [dict[@"dt"] intValue];
    _dRId = dict[@"dRId"] ?: @"";
    _pn = dict[@"pn"];
    _auId = dict[@"auId"];
    _n = dict[@"n"];
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [@{
        @"cId": @(_cId),
        @"aId": _aId ?: @"",
        @"dt": @(_dt),
        @"dRId": _dRId ?: @""
    } mutableCopy];
    if (_pn) dict[@"pn"] = _pn;
    if (_auId) dict[@"auId"] = _auId;
    if (_n) dict[@"n"] = _n;
    return dict;
}

@end

#pragma mark - PushAppInstallRes

@implementation PushAppInstallRes

- (instancetype)init {
    _cId = APP_INSTALL_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    _cId = [dict[@"cId"] intValue];
    _aId = dict[@"aId"] ?: @"";
    _rc = [dict[@"rc"] intValue];
    _cmt = dict[@"cmt"];
    _usRid = dict[@"usRid"] ?: @"";
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [@{
        @"cId": @(_cId),
        @"aId": _aId ?: @"",
        @"rc": @(_rc),
        @"usRid": _usRid ?: @""
    } mutableCopy];
    if (_cmt) dict[@"cmt"] = _cmt;
    return dict;
}

@end


@implementation PushAppUnregUserReq

- (instancetype)init {
    self.cId = APP_UNREG_USER_CMD_ID;
    return self;
}

- (NSDictionary *)toDictionary {
    return @{
        @"cId": @(self.cId),
        @"aId": self.aId ?: @"",
        @"dRId": self.dRId ?: @""
    };
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    PushAppUnregUserReq *obj = [PushAppUnregUserReq new];
    obj.cId = [dict[@"cId"] intValue];
    obj.aId = dict[@"aId"];
    obj.dRId = dict[@"dRId"];
    return obj;
}

@end


@implementation PushAppUnregUserRes

- (instancetype)init {
    self.cId = APP_UNREG_USER_CMD_ID;
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *json = [@{
        @"cId": @(self.cId),
        @"aId": self.aId ?: @"",
        @"rc": @(self.rc),
        @"dRId": self.dRId ?: @""
    } mutableCopy];
    if (self.cmt != nil) {
        json[@"cmt"] = self.cmt;
    }
    return json;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    PushAppUnregUserRes *obj = [PushAppUnregUserRes new];
    obj.cId = [dict[@"cId"] intValue];
    obj.aId = dict[@"aId"];
    obj.rc = [dict[@"rc"] intValue];
    obj.cmt = dict[@"cmt"];
    obj.dRId = dict[@"dRId"];
    return obj;
}

@end



@implementation PushAppMsgReadNoti

- (instancetype)init {
    self = [super init];
    self.cId = APP_MSG_READ_NOTI_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    self.cId = [dict[@"cId"] intValue];
    self.aId = dict[@"aId"];
    self.dRId = dict[@"dRId"];
    self.mId = dict[@"mId"];
    return self;
}

- (NSDictionary *)toDictionary {
    return @{
        @"cId": @(self.cId),
        @"aId": self.aId,
        @"dRId": self.dRId,
        @"mId": self.mId
    };
}

@end


@implementation PushAppMsgInfoReq

- (instancetype)init {
    self = [super init];
    self.cId = APP_MSG_INFO_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    self.cId = [dict[@"cId"] intValue];
    self.aId = dict[@"aId"];
    self.dRId = dict[@"dRId"];
    self.mId = dict[@"mId"];
    return self;
}

- (NSDictionary *)toDictionary {
    return @{
        @"cId": @(self.cId),
        @"aId": self.aId,
        @"dRId": self.dRId,
        @"mId": self.mId
    };
}

@end


@implementation PushAppMsgInfoRes

- (instancetype)init {
    self = [super init];
    self.cId = APP_MSG_INFO_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    self.cId = [dict[@"cId"] intValue];
    self.aId = dict[@"aId"];
    self.rc = [dict[@"rc"] intValue];
    self.cmt = dict[@"cmt"];
    self.mId = dict[@"mId"];
    self.ast = [dict[@"ast"] longLongValue];
    self.ase = [dict[@"ase"] intValue];
    self.mst = [dict[@"mst"] longLongValue];
    self.mt = [dict[@"mt"] intValue];
    self.ms = [dict[@"ms"] intValue];
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [@{
        @"cId": @(self.cId),
        @"aId": self.aId,
        @"rc": @(self.rc),
        @"mId": self.mId,
        @"ast": @(self.ast),
        @"ase": @(self.ase),
        @"mst": @(self.mst),
        @"mt": @(self.mt),
        @"ms": @(self.ms)
    } mutableCopy];

    if (self.cmt != nil) {
        dict[@"cmt"] = self.cmt;
    }

    return dict;
}

@end


@implementation PushAppImgDataReq

- (instancetype)init {
    self = [super init];
    self.cId = APP_IMG_DATA_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    self.cId = [dict[@"cId"] intValue];
    self.aId = dict[@"aId"];
    self.mId = dict[@"mId"];
    self.iId = dict[@"iId"];
    return self;
}

- (NSDictionary *)toDictionary {
    return @{
        @"cId": @(self.cId),
        @"aId": self.aId,
        @"mId": self.mId,
        @"iId": self.iId
    };
}

@end


@implementation PushAppImgDataRes

- (instancetype)init {
    self = [super init];
    self.cId = APP_IMG_DATA_CMD_ID;
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    self.cId = [dict[@"cId"] intValue];
    self.aId = dict[@"aId"];
    self.rc = [dict[@"rc"] intValue];
    self.cmt = dict[@"cmt"];
    self.imgD = dict[@"imgD"];
    return self;
}

- (NSDictionary *)toDictionary {
    NSMutableDictionary *dict = [@{
        @"cId": @(self.cId),
        @"aId": self.aId,
        @"rc": @(self.rc),
        @"imgD": self.imgD
    } mutableCopy];

    if (self.cmt != nil) {
        dict[@"cmt"] = self.cmt;
    }

    return dict;
}

@end

File: device-app/ios-swift/helloUms/bridgeUms/PushAppProto.swift
class PushAppProto {

    class PushAppAuthNumberReq: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
        // account id
        var aId: String = ""
        // country code
        var cc: String = ""
        // phone number
        var pn: String = ""
        
        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
        }
        
        init(aId: String, cc: String, pn: String) {
            self.cId = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
            self.aId = aId
            self.cc = cc
            self.pn = pn
        }
    }


    class PushAppAuthNumberRes: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
        // account id
        var aId: String = ""
        // result code
        var rc: Int = 0
        // comment
        var cmt: String? = nil

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
        }
        
        init(aId: String, rc: Int, cmt: String) {
            self.cId = UmsProtocol.APP_AUTH_NUMBER_CMD_ID
            self.aId = aId
            self.rc = rc
            self.cmt = cmt
        }
    }

    class PushAppVerifyAuthNumberReq: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
        // account id
        var aId: String = ""
        // device type  1: android, 2: ios
        var dt: Int = 0
        // phone number
        var pn: String = ""
        // auth number
        var an: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
        }
        
        init(aId: String, dt: Int, pn: String, an: String) {
            self.cId = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
            self.aId = aId
            self.dt = dt
            self.pn = pn
            self.an = an
        }

    }

    class PushAppVerifyAuthNumberRes: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
        // account id
        var aId: String = ""
        // result code
        var rc: Int = 0
        // comment
        var cmt: String? = nil

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
        }
        
        init(aId: String, rc: Int, cmt: String) {
            self.cId = UmsProtocol.APP_VERITY_AUTH_NUMBER_CMD_ID
            self.aId = aId
            self.rc = rc
            self.cmt = cmt
        }

    }

    class PushAppInstallReq: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_INSTALL_CMD_ID
        // account id
        var aId: String = ""
        // device type  1: android, 2: ios
        var dt: Int = 0
        // device registration id
        var dRId: String = ""
        // phone number
        var pn: String? = nil
        // app user id
        var auId: String? = nil
        // name
        var n: String? = nil

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_INSTALL_CMD_ID
        }
        
        init(aId: String, dt: Int, dRId: String, pn: String, auId: String?, n: String?) {
            self.cId = UmsProtocol.APP_INSTALL_CMD_ID
            self.aId = aId
            self.dt = dt
            self.dRId = dRId
            self.pn = pn
            self.auId = auId
            self.n = n
        }

    }

    class PushAppInstallRes: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_INSTALL_CMD_ID
        // account id
        var aId: String = ""
        // result code
        var rc: Int = 0
        // comment
        var cmt: String? = nil
        // ums server registration id
        var usRid: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_INSTALL_CMD_ID
        }
        
        init(aId: String, rc: Int, cmt: String, usRid: String) {
            self.cId = UmsProtocol.APP_INSTALL_CMD_ID
            self.aId = aId
            self.rc = rc
            self.cmt = cmt
            self.usRid = usRid
        }

    }


    class PushAppUnregUserReq: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_UNREG_USER_CMD_ID
        // account id
        var aId: String = ""
        // device registration id
        var dRId: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_UNREG_USER_CMD_ID
        }
        
        init(aId: String, dRId: String) {
            self.cId = UmsProtocol.APP_UNREG_USER_CMD_ID
            self.aId = aId
            self.dRId = dRId
        }

    }


    class PushAppUnregUserRes: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_UNREG_USER_CMD_ID
        // account id
        var aId: String = ""
        // result code
        var rc: Int = 0
        // comment
        var cmt: String? = nil
        // device registration id
        var dRId: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_UNREG_USER_CMD_ID
        }
        
        init(aId: String, rc: Int, cmt: String, dRId: String) {
            self.cId = UmsProtocol.APP_UNREG_USER_CMD_ID
            self.aId = aId
            self.rc = rc
            self.cmt = cmt
            self.dRId = dRId
        }

    }

    class PushAppMsgReadNoti: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_MSG_READ_NOTI_CMD_ID
        // account id
        var aId: String = ""
        // device registration id
        var dRId: String = ""
        // message id
        var mId: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_MSG_READ_NOTI_CMD_ID
        }
        
        init(aId: String, dRId: String, mId: String) {
            self.cId = UmsProtocol.APP_MSG_READ_NOTI_CMD_ID
            self.aId = aId
            self.dRId = dRId
            self.mId = mId
        }

    }

    class PushAppMsgInfoReq: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_MSG_INFO_CMD_ID
        // account id
        var aId: String = ""
        // device registration id
        var dRId: String = ""
        // message id
        var mId: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_MSG_INFO_CMD_ID
        }
        
        init(aId: String, dRId: String, mId: String) {
            self.cId = UmsProtocol.APP_MSG_INFO_CMD_ID
            self.aId = aId
            self.dRId = dRId
            self.mId = mId
        }

    }

    class PushAppMsgInfoRes: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_MSG_INFO_CMD_ID
        // account id
        var aId: String = ""
        // result code
        var rc: Int = 0
        // comment
        var cmt: String? = nil
        // message id
        var mId: String = ""
        // alimtalk send time (알림톡 발신 시간)
        var ast: Int64 = 0
        // alimtalk state 알림톡 전달상태 (0: 미발신, 1: 발신요청, 2: 전달성공, 3: 전달실패)
        var ase: Int = 0
        // munja send time (문자 발신 시간)
        var mst: Int64 = 0
        // munja type (11: sms, 12: lms, 13: mms)
        var mt: Int = 0
        // munja state (0: 미발신, 1: 발신요청, 2: 전달성공, 3: 전달실패)
        var ms: Int = 0

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_MSG_INFO_CMD_ID
        }
        
        init(aId: String, rc: Int, cmt: String, mId: String,
                           ast: Int64, ase: Int, mst: Int64, mt: Int, ms: Int) {
            self.cId = UmsProtocol.APP_MSG_INFO_CMD_ID
            self.aId = aId
            self.rc = rc
            self.cmt = cmt
            self.mId = mId
            self.ast = ast
            self.ase = ase
            self.mst = mst
            self.mt = mt
            self.ms = ms
        }

    }

    class PushAppImgDataReq: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_IMG_DATA_CMD_ID
        // account id
        var aId: String = ""
        // message id
        var mId: String = ""
        // image id
        var iId: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_IMG_DATA_CMD_ID
        }
        
        init(aId: String, mId: String, iId: String,) {
            self.cId = UmsProtocol.APP_IMG_DATA_CMD_ID
            self.aId = aId
            self.mId = mId
            self.iId = iId
        }

    }

    class PushAppImgDataRes: Codable {
        // command id
        var cId: Int = UmsProtocol.APP_IMG_DATA_CMD_ID
        // account id
        var aId: String = ""
        // result code
        var rc: Int = 0
        // comment
        var cmt: String? = nil
        // image data (base64 인코딩 포맷)
        var imgD: String = ""

        // 기본 생성자
        init() {
            self.cId = UmsProtocol.APP_IMG_DATA_CMD_ID
        }
        
        init(aId: String, rc: Int, cmt: String, imgD: String) {
            self.cId = UmsProtocol.APP_IMG_DATA_CMD_ID
            self.aId = aId
            self.rc = rc
            self.cmt = cmt
            self.imgD = imgD
        }

    }
}
